core/src/Kokkos_CopyViews.hpp

```diff
@@ -3095,11 +2993,13 @@ inline void resize(Kokkos::View<T, P...>& v,
   static_assert(Kokkos::ViewTraits<T, P...>::is_managed,
                 "Can only resize managed views");
 
-  view_type v_resized(v.label(), layout);
+  if (v.layout() != layout) {
```

Comment:
Hm this is problematic since there are custom layouts in circulation.

Reply:
Yes, that was also my concern a little bit. I added it because @dalg24 asked for it and checked that it works as expected for at least `LayoutRight` but I would tend to exclude it from this pull request.